cms/djangoapps/contentstore/views/item.py

```diff
@@ -427,13 +427,16 @@ def xblock_view_handler(request, usage_key_string, view_name):
 
             # Note that the container view recursively adds headers into the preview fragment,
             # so only the "Pages" view requires that this extra wrapper be included.
+            display_label = xblock.display_name or xblock.scope_ids.block_type
+            if not xblock.display_name and xblock.scope_ids.block_type == 'html':
+                display_label = _("Text")
```

Comment:
I'm thinking this logic is a little hard to read. Maybe something like:

```suggestion
                display_label = ''
                if display_name in {"", None}:  // if empty, default to block type
                       if xblock.scope_ids.block_type == 'html':  // some comment here about the ticket.
                          display_label = _("Text")
                       else: 
                           xblock.scope_ids.block_type
                else: 
                    display_label = xblock.display_name
```

Does that make more sense to you? I think it does to me